test_feedforward.py and test_block_factory.py fail due to wrong pytorch distributed backend (should be GLOO, is set to NCLL)

# 🐛 Bug

test_feedforward.py and test_block_factory.py fail pytest

FAILED test_block_factory.py::test_xformer_decoder_block[device0-deepnorm-MixtureOfExperts-visual-True-smelu-2-False-0.1-0.1] - RuntimeError: Distributed package doesn't have NCCL built in
FAILED test_feedforward.py::test_moe[get_expert-4-top_2] - RuntimeError: Distributed package doesn't have NCCL built in

## Command

pytest test_feedforward.py

pytest test_block_factory.py

note including patch fix in tracker

--- a/xformers/helpers/test_utils.py
+++ b/xformers/helpers/test_utils.py
@@ -4,14 +4,15 @@
 # LICENSE file in the root directory of this source tree.
 
 
+import os
 import tempfile
 
 import torch
 
-import os
 is_windows = False
-if (os.environ.get('OS','') == 'Windows_NT'): #pytorch on windows uses gloo not ncll
+if (os.environ.get('OS', '') == 'Windows_NT'):  # pytorch on windows uses gloo not ncll
     is_windows = True
+
 
 def init_torch_distributed_local():
     if torch.distributed.is_initialized():

--- a/xformers/helpers/test_utils.py
+++ b/xformers/helpers/test_utils.py
@@ -10,7 +10,7 @@ import tempfile
 import torch
 
 is_windows = False
-if (os.environ.get('OS', '') == 'Windows_NT'):  # pytorch on windows uses gloo not ncll
+if os.environ.get("OS", "") == "Windows_NT":  # pytorch on windows uses gloo not ncll
     is_windows = True
 
 

--- a/xformers/helpers/test_utils.py
+++ b/xformers/helpers/test_utils.py
@@ -4,13 +4,13 @@
 # LICENSE file in the root directory of this source tree.
 
 
-import os
+import sys
 import tempfile
 
 import torch
 
 is_windows = False
-if os.environ.get("OS", "") == "Windows_NT":  # pytorch on windows uses gloo not ncll
+if sys.platform == "win32":  # pytorch on windows uses gloo not ncll
     is_windows = True
 
 

--- a/xformers/helpers/test_utils.py
+++ b/xformers/helpers/test_utils.py
@@ -8,6 +8,10 @@ import tempfile
 
 import torch
 
+import os
+is_windows = False
+if (os.environ.get('OS','') == 'Windows_NT'): #pytorch on windows uses gloo not ncll
+    is_windows = True
 
 def init_torch_distributed_local():
     if torch.distributed.is_initialized():
@@ -16,7 +20,7 @@ def init_torch_distributed_local():
     init_url = "file://" + tempfile.mkstemp()[1]
     backend = (
         torch.distributed.Backend.NCCL
-        if torch.cuda.is_available()
+        if torch.cuda.is_available() and not is_windows
         else torch.distributed.Backend.GLOO
     )
     torch.distributed.init_process_group(

